### ISSUE TITLE
git support in kpt pkg pull and push

### ISSUE THREAD

**Issue by loudej** [kptdev/kpt#2612]
_This feature request assumes OCI support, and pkg pull and push commands exist._

`kpt pkg pull` and `kpt pkg push` should support git locations in addition to oci.

A pkg pull from a git source would be used copy a `repo/path@version` coordinate to a local folder unchanged (without the pkg get downstreaming logic)

A pkg push to a git source would be used to do the reverse. Starting from a local folder (not as part of an entire repo clone) create and push a single commit to the named repo and path onto the @version reference.

The interpretation of @version can be a bit ambiguous. Simplest interpretation could be to use branch names like @main or other feature branches as the target. If push is meant to create new branch or version a question would be what commit to use as the base...

Unclear if this is necessary given git cli is understood - but adding a feature request to follow-up on question raised in the OCI design doc.

**Comment by mikebz** [comment]
is `kpt pkg pull` different than `kpt pkg get`? 
If so what is the difference between pull and get?

https://kpt.dev/reference/cli/pkg/get/
```
kpt pkg get REPO_URI[.git]/PKG_PATH[@VERSION] [LOCAL_DEST_DIRECTORY] [flags]
```

**Comment by loudej** [comment]
Yes, pkg get creates a new package, changing the name in the result, and changes upstream data to point at the source of the get.

Kpt pkg pull and push on the other hand don't create a new package name, or change contents in any way. They're just pulling files from image to folder, and pushing from folder to image.

Pull/push are needed for editing. They can also be used to move packages between registriea. Another example is configsync would use pkg pull. 

https://github.com/loudej/kpt/blob/oci-design/docs/design-docs/01-oci-support.md#comparison-of-pkg-get-and-pkg-pull

See also, various usages

https://gist.github.com/loudej/a21df2a3e3b7ba3b5a01ef94aba9b14a
